Content.Server/Storage/EntitySystems/EntityStorageSystem.cs

```diff
@@ -59,12 +114,239 @@ public void EmptyContents(EntityUid uid, EntityStorageComponent? component = nul
         {
             if (component.Contents.Remove(contained))
             {
-                Transform(contained).WorldPosition = component.ContentsDumpPosition();
+                Transform(contained).WorldPosition = Transform(uid).WorldPosition;
                 if (TryComp(contained, out IPhysBody? physics))
                 {
                     physics.CanCollide = true;
                 }
             }
         }
     }
+
+    public void OpenStorage(EntityUid uid, EntityStorageComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        component.Open = true;
+        EmptyContents(uid, component);
+        ModifyComponents(uid, component);
+        SoundSystem.Play(component.OpenSound.GetSound(), Filter.Pvs(component.Owner), component.Owner);
+    }
+
+    public void CloseStorage(EntityUid uid, EntityStorageComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        component.Open = false;
+
+        var ev = new StorageBeforeCloseEvent(uid, _lookup.GetEntitiesInRange(uid, component.EnteringRange, LookupFlags.Approximate));
+        RaiseLocalEvent(uid, ev, true);
+        
+        var count = 0;
+        foreach (var entity in ev.Contents)
+        {
+            if (_container.IsEntityInContainer(entity))
+                continue;
```

Comment:
Changes were made to how lookups works, so now by default they contained entities, unless they are specifically requested via `LookupFlags.Contained`. So this check is unnecessary now.